src/Components/Server/src/DependencyInjection/ConfigureStaticFilesOptions.cs

```diff
@@ -0,0 +1,116 @@
+            options = options ?? throw new ArgumentNullException(nameof(options));
+
+            if (name != Options.DefaultName)
+            {
+                return;
+            }
+
+            // Basic initialization in case the options weren't initialized by any other component
+            options.ContentTypeProvider = options.ContentTypeProvider ?? new FileExtensionContentTypeProvider();
+            if (options.FileProvider == null && Environment.WebRootFileProvider == null)
+            {
+                throw new InvalidOperationException("Missing FileProvider.");
+            }
+
+            options.FileProvider = options.FileProvider ?? Environment.WebRootFileProvider;
+
+            var prepareResponse = options.OnPrepareResponse;
+            if (prepareResponse == null)
+            {
+                options.OnPrepareResponse = BlazorApplicationBuilderExtensions.SetCacheHeaders;
+            }
+            else
+            {
+                void PrepareResponse(StaticFileResponseContext context)
+                {
+                    prepareResponse(context);
+                    BlazorApplicationBuilderExtensions.SetCacheHeaders(context);
+                }
+
+                options.OnPrepareResponse = PrepareResponse;
+            }
+
+            // Add our provider
+            var provider = new ManifestEmbeddedFileProvider(typeof(ConfigureStaticFilesOptions).Assembly);
+
+            options.FileProvider = new CompositeFileProvider(provider, new ContentReferencesFileProvider(), options.FileProvider);
+        }
+
+        private class ContentReferencesFileProvider : IFileProvider
+        {
+            byte[] _data = Encoding.UTF8.GetBytes(@"{ ""cssReferences"": [], ""jsReferences"": [] }");
+
+            public IDirectoryContents GetDirectoryContents(string subpath)
+            {
+                return new NotFoundDirectoryContents();
+            }
+
+            public IFileInfo GetFileInfo(string subpath)
+            {
+                if (subpath.Equals("/_framework/blazor.boot.json", StringComparison.OrdinalIgnoreCase))
+                {
+                    return new MemoryFileInfo(_data);
+                }
+
+                return new NotFoundFileInfo(subpath);
+            }
+
+            public IChangeToken Watch(string filter) => NullChangeToken.Singleton;
+
+            private class MemoryFileInfo : IFileInfo
+            {
+                private readonly byte[] _data;
+
+                public MemoryFileInfo(byte[] data)
+                {
+                    _data = data;
+                }
+
+                public bool Exists => true;
+
+                public long Length => _data.Length;
+
+                public string PhysicalPath => "/_framework/blazor.boot.json";
+
+                public string Name => "blazor.boot.json";
+
+                public DateTimeOffset LastModified => DateTimeOffset.FromUnixTimeSeconds(0);
```

Comment:
`DateTimeOffset.MinValue`?

Reply:
what's the point. This is January 1st 1970 and it's a well known value.